### ISSUE TITLE
NW.js Linux export creates bloated invalid ZIP file

### ISSUE THREAD

**Issue by johnBrink** [Scirra/Construct-bugs#6996]
## Problem description

I cannot export my game for Linux. When I do, the ZIP file is corrupted and is about 200 MB (expected around 60-70). On Windows, I can open up the archive with 7-Zip and extract the data, minus the Linux permissions. On Linux, the archive cannot be opened.

## Attach a .c3p

[linuxtest.zip](https://github.com/Scirra/Construct-bugs/files/11362706/linuxtest.zip)

## Steps to reproduce

1. Create a new project. Put a single Text object into the initial layout.
2. Export to NW.js
3. First page of export wizard: Default options (deduplicate, all WebP, recompress, no minify)
4. Second page: Latest NW.js (0.75.0), platform Linux64, package assets, compress final zip, window frame, resizable, ignore GPU blacklist, with or without export for Steam
5. Run the export and download the file. It should take a long time and create an unusually large ZIP

![export1](https://user-images.githubusercontent.com/7257437/235382008-fd7e5bed-308c-4e90-b3e8-2c52940fc0e4.png)

![export2](https://user-images.githubusercontent.com/7257437/235382013-c5906893-f48b-416d-8702-cd25b39d1de9.png)

## Observed result

The ZIP file will be corrupted. In 7-Zip for Windows, you will be unable to add or modify records in the archive. In Engrampa Archive Manager for Linux, opening fails with "Headers Error".

## Expected result

The archive should open successfully, allowing me to extract my game and play it on my Linux machine.

## More details

I did this using the Chrome webapp version of Construct 3, in both Windows and Linux. I also tried Firefox on Linux with the same results.

**Affected browsers/platforms:** Windows 10, Xubuntu Linux, Chrome 112.0.5615.138, Firefox 110.0.1

**First affected release:** Noticed as early as v317 - it had worked at some point but not sure when.

## System details

<details><summary>View details</summary>

Platform information
Product: Construct 3 r336 (stable)
Browser: Chrome 112.0.5615.165
Browser engine: Chromium
Context: webapp
Operating system: Linux 5.15.0
Device type: desktop
Device pixel ratio: 1
Logical CPU cores: 4
Approx. device memory: 8 GB
User agent: Mozilla/5.0 (X11; Linux x86_64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/112.0.0.0 Safari/537.36
Language setting: en-US

Local storage
Storage quota (approx): 57 gb
Storage usage (approx): 4.1 gb (7.1%)
Persistant storage: No

Browser support notes
This list contains missing features that are not required, but could improve performance or user experience if supported.

WebGL indicates a major performance caveat. It is probably using software rendering.
WebGL information
Version string: WebGL 2.0 (OpenGL ES 3.0 Chromium)
Numeric version: 2
Supports NPOT textures: yes
Supports GPU profiling: no
Supports highp precision: yes
Vendor: Google Inc. (Google)
Renderer: ANGLE (Google, Vulkan 1.3.0 (SwiftShader Device (Subzero) (0x0000C0DE)), SwiftShader driver)
Major performance caveat: yes
Maximum texture size: 8192
Point size range: 1 to 1023
Extensions:

EXT_color_buffer_float
EXT_color_buffer_half_float
EXT_float_blend
EXT_texture_compression_bptc
EXT_texture_compression_rgtc
EXT_texture_filter_anisotropic
OES_draw_buffers_indexed
OES_texture_float_linear
OVR_multiview2
WEBGL_compressed_texture_astc
WEBGL_compressed_texture_etc
WEBGL_compressed_texture_etc1
WEBGL_compressed_texture_s3tc
WEBGL_compressed_texture_s3tc_srgb
WEBGL_debug_renderer_info
WEBGL_lose_context
WEBGL_multi_draw
Audio information
System sample rate: 44100 Hz
Output channels: 2
Output interpretation: speakers
Supported decode formats:

WebM Opus (audio/webm; codecs=opus)
Ogg Opus (audio/ogg; codecs=opus)
WebM Vorbis (audio/webm; codecs=vorbis)
Ogg Vorbis (audio/ogg; codecs=vorbis)
MPEG-4 AAC (audio/mp4; codecs=mp4a.40.5)
MP3 (audio/mpeg)
FLAC (audio/flac)
PCM WAV (audio/wav; codecs=1)
Supported encode formats:

WebM Opus (audio/webm; codecs=opus)
Video information
Supported decode formats:

WebM AV1 (video/webm; codecs=av01.0.00M.08)
MP4 AV1 (video/mp4; codecs=av01.0.00M.08)
WebM VP9 (video/webm; codecs=vp9)
WebM VP8 (video/webm; codecs=vp8)
Ogg Theora (video/ogg; codecs=theora)
H.264 (video/mp4; codecs=avc1.42E01E)
Supported encode formats:

WebM VP9 (video/webm; codecs=vp9)
WebM VP8 (video/webm; codecs=vp8)

</details>

**Comment by johnBrink** [comment]
Update: I should also point out that the Windows export also has a slightly different issue.
When I open the game export ZIP in Windows it extracts without problem, but package.nw has a header error in this case, and I cannot make changes to it (like to add the Steam library files) unless I create a new ZIP file and copy the files from it.

When I try to add files to package.nw in 7-Zip, I get an alert saying "Not implemented".

Also - on further investigation, it seems like the file size may be normal for latest versions of NW.js. If that's the case, then the invalid ZIP headers would seem to be the only issue.

**Comment by AshleyScirra** [comment]
This all seems to be working fine for me - I checked on both a Windows and Ubuntu system. The zip downloads OK, is the correct size (~137mb), and opens and extracts OK, as well as package.nw if renamed to .zip. So I'm not seeing any of the reported issues on either system.

My best guess is either it's a connection problem and the download is somehow being corrupted, or your zip software is way out of date and doesn't support the Zip64 format, which we switched to using a while back. Both the built-in Windows zip reader and Ubuntu's Archive Manager had no problem with these zip files.
